Increase documentation about new features at Spec and readme (#295)
jakartaee/data: README.adoc

=== Repository

Jakarta Data provides repository abstractions designed to streamline the development of data access layers for various persistent stores, eliminating the need for extensive boilerplate code. These repository abstractions are broadly classified into two categories:

> * *Built-in Repository Interfaces*: Jakarta Data offers a hierarchy of built-in repository interfaces. The root of this hierarchy is the `DataRepository` interface, which includes several specialized interfaces like `CrudRepository`, `KeyValueRepository`, and more. These interfaces are designed to handle common data access operations and provide a solid foundation for your data layer.

Reviewer: Jakarta Data doesn't have any `KeyValueRepository`.
```suggestion
* *Built-in Repository Interfaces*: Jakarta Data offers a hierarchy of built-in repository interfaces. The root of this hierarchy is the `DataRepository` interface, which includes several specialized interfaces like `CrudRepository`, `BasicRepository`, and more. These interfaces are designed to handle common data access operations and provide an extensible foundation for your data layer.
```

Author: Sorry, my heart is on NoSQL databases, even by accident.
Thank you for that.

repository.save(ferrari);
----

* *Custom Repository Interfaces*: Besides the built-in repository interfaces, Jakarta Data enables you to create custom repository interfaces tailored to your domain requirements. These custom interfaces serve as a bridge between your domain's semantics and the underlying database operations. With annotations like `Insert`, `Update`, `Delete`, and `Save`, you can define expressive and domain-specific repository methods that align seamlessly with your application's needs.

[source,java]
----
@Repository
public interface Garage {

>   @Save
>   Car park(Car car);

Reviewer: `Insert` is better aligned with `Save` for a method that is named `park`.  It doesn't make sense to refer to updating the state of a Car (maybe its license plate number) as parking it.

```suggestion
  @Insert
  Car park(Car car);
```

}
----

> ==== Type-safe Access to Entity Attributes
> 
> Jakarta Data provides a static metamodel that allows entity attributes to be accessed by applications in a type-safe manner. This metamodel generates metadata classes that mirror your entity classes, enabling you to work with entity attributes using compile-time safety.
> 
> With the Jakarta Data static metamodel, you can access entity attributes in a type-safe manner. Here's a quick example to demonstrate how it works:
> 
> [source,java]
> ----
> 
> @Entity
> public class Product {
>   public long id;
>   public String name;
>   public float price;
> }
> ----
> 
> 
> Usage example:
> 
> [source,java]
> ----
> 
> List<Product> found = products.findByNameContains(searchPattern,
>                                                   Product_.price.desc(),
>                                                   Product_.name.asc(),
>                                                   Product_.id.asc());
> ----
> 
> 
> In this example, `Product_` is the metadata class generated by Jakarta Data, allowing you to access entity attributes in a type-safe way. It helps prevent runtime errors related to attribute names and enables more reliable and efficient data access.
> 
> 

Reviewer: There is a separate issue under which discussion is occurring about whether we want to change the StaticMetamodel.  We should avoid adding new sections like this while that is still being discussed in the other issue.  Also, this document already contains a section that is titled "Type-safe Access to Entity Attributes", and so we should consider putting the changes there when we do make them.

```suggestion
```
